packages/core/src/internal/types/journal.ts

```diff
@@ -49,4 +49,9 @@ export interface ICommandJournal {
    * Read out the stored deployment update commands.
    */
   read(): AsyncGenerator<DeployStateExecutionCommand, void, unknown>;
+  readAll(): AsyncGenerator<
+    DeployStateExecutionCommand & { chainId: number },
```

Comment:
This doesn't need changed, but I suspect it means we should rework the structure of the journal (probably multiple files - one per chain).